src/aks-preview/azext_aks_preview/azuremonitorprofile.py

```diff
@@ -0,0 +1,812 @@
+    # check if region supports DCR and DCRA
+    for _ in range(3):
+        try:
+            feature_check_url = f"https://management.azure.com/subscriptions/{mac_subscription_id}/providers/Microsoft.Insights?api-version=2020-10-01"
+            r = send_raw_request(cmd.cli_ctx, "GET", feature_check_url)
+            error = None
+            break
+        except CLIError as e:
+            error = e
+    else:
+        raise error
+    json_response = json.loads(r.text)
+    for resource in json_response["resourceTypes"]:
+        region_ids = map(lambda x: region_names_to_id[x],
+                         resource["locations"])  # map is lazy, so doing this for every region isn"t slow
+        if resource["resourceType"].lower() == "datacollectionrules" and mac_location not in region_ids:
+            raise ClientRequestError(
+                f"Data Collection Rules are not supported for MAC region {mac_location}")
+        elif resource[
+                "resourceType"].lower() == "datacollectionruleassociations" and cluster_region not in region_ids:
+            raise ClientRequestError(
+                f"Data Collection Rule Associations are not supported for cluster region {cluster_region}")
+    return mac_location
+
+def create_dce(cmd, cluster_subscription, cluster_resource_group_name, cluster_name, mac_region):
+    print("Calling function create_dce")
+    from azure.cli.core.util import send_raw_request
+
+    dce_name = get_default_dce_name(mac_region, cluster_name)
+
+    dce_resource_id = "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Insights/dataCollectionEndpoints/{2}".format(
+            cluster_subscription,
+            cluster_resource_group_name,
+            dce_name,
+        )
+
+    for _ in range(3):
+        try:
+            dce_url = f"https://management.azure.com{dce_resource_id}?api-version={DC_API}"
+            dce_creation_body = json.dumps({"name": dce_name,
+                                            "location": mac_region,
+                                            "kind": "Linux",
+                                            "properties": {}})
+            send_raw_request(cmd.cli_ctx, "PUT",
+                             dce_url, body=dce_creation_body)
+            error = None
+            return dce_resource_id
+        except CLIError as e:
+            error = e
+    else:
+        raise error
+
+def create_dcr(cmd, mac_region, azure_monitor_workspace_resource_id, cluster_subscription, cluster_resource_group_name, cluster_name, dce_resource_id):
+    from azure.cli.core.util import send_raw_request
+
+    print("Calling function create_dcr")
+
+    dcr_name = get_default_dcr_name(mac_region, cluster_name)
+    dcr_resource_id = "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Insights/dataCollectionRules/{2}".format(
+        cluster_subscription,
+        cluster_resource_group_name,
+        dcr_name
+    )
+
+    dcr_creation_body = json.dumps({"location": mac_region,
```

Comment:
This seems somehow similar to the code in [azure-cli/acs](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py#L388), not sure if it's possible to reuse the code.

Reply:
Since we've done a lot of bug bashes and testing on this code for the release I'm thinking to leave this as is for now since we're closing in on the public preview date.

I have created a task [here](https://msazure.visualstudio.com/InfrastructureInsights/_workitems/edit/15524100) in my user story in azure devops for GA related tasks (For e.g. the other one on there is moving the recording rules from a storage account to the Metrics RP location when its available) and plan on doing it at that time. Let me know if this sounds good to you?